src/main/python/afp_cli/aws_credentials_file.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from six.moves import configparser
+import os
+import six
+
+def write(aws_credentials, filename=None, profile_name='default'):
+
+    if six.PY2:
+        # WTF
+        ORIG_DEFAULTSECT = configparser.DEFAULTSECT
+        configparser.DEFAULTSECT = 'default'
+
+    try:
+
+        if not filename:
+            filename = os.path.expanduser("~") + '/.aws/credentials'
+
+        if not os.path.exists(os.path.dirname(filename)):
+            os.makedirs(os.path.dirname(filename))
+
+        config = configparser.RawConfigParser()
+        config.read(filename)
+
+        if not config.has_section(profile_name) and (profile_name.lower() != 'default' or six.PY3):
+            config.add_section(profile_name)
+
+        config.set(profile_name, 'aws_access_key_id', aws_credentials['AWS_ACCESS_KEY_ID'])
+        config.set(profile_name, 'aws_secret_access_key', aws_credentials['AWS_SECRET_ACCESS_KEY'])
+        config.set(profile_name, 'aws_session_token', aws_credentials['AWS_SESSION_TOKEN'])
```

Comment:
+1 for merge now and improve later.

On 14 October 2015 14:33:37 CEST, "Oliver Schönherr" <notifications@github.com> wrote:
>> +
>> +        if not filename:
>> +            filename = os.path.expanduser("~") + '/.aws/credentials'
>> +
>> +        if not os.path.exists(os.path.dirname(filename)):
>> +            os.makedirs(os.path.dirname(filename))
>> +
>> +        config = configparser.RawConfigParser()
>> +        config.read(filename)
>> +
>> +        if not config.has_section(profile_name) and
>(profile_name.lower() != 'default' or six.PY3):
>> +            config.add_section(profile_name)
>> +
>> +        config.set(profile_name, 'aws_access_key_id',
>aws_credentials['AWS_ACCESS_KEY_ID'])
>> +        config.set(profile_name, 'aws_secret_access_key',
>aws_credentials['AWS_SECRET_ACCESS_KEY'])
>> +        config.set(profile_name, 'aws_session_token',
>aws_credentials['AWS_SESSION_TOKEN'])
>
>@schlomo in my case i don't need this information in the file. Maybe we
>should print this information to the console.
>
>But, we should first accept this pull request in the current state, and
>discuss further improvements in a later issue. Because this feature is
>very helpful for some common development workflows.
>
>
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/ImmobilienScout24/afp-cli/pull/17/files#r41987103

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.